Bump watchdog to 6.0.0

mweinelt opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Update watchdog from 2.3.1 (Feb 2023) to 6.0.0 (Nov 2024).

[Changelog](https://github.com/gorakhargosh/watchdog/blob/v6.0.0/changelog.rst)

https://github.com/gorakhargosh/watchdog/compare/v5.0.3...v6.0.0
https://github.com/gorakhargosh/watchdog/compare/v5.0.2...v5.0.3
https://github.com/gorakhargosh/watchdog/compare/v5.0.1...v5.0.2
https://github.com/gorakhargosh/watchdog/compare/v5.0.0...v5.0.1
https://github.com/gorakhargosh/watchdog/compare/v4.0.2...v5.0.0 (contains the [typing change](https://github.com/gorakhargosh/watchdog/commit/84d5adb8e937a4bce55f73890e3c53d26ecd231a) and enforces [keyword arguments](https://github.com/gorakhargosh/watchdog/commit/b87e645c010afa1658ea66a13f8bcc5662a543d0#diff-d94873adedb3d85744a8e4225b200bf49e92e5c61417ea15f33b403d16920546R298))
https://github.com/gorakhargosh/watchdog/compare/v4.0.1...v4.0.2 (adds Python 3.13 support)
https://github.com/gorakhargosh/watchdog/compare/v4.0.0...v4.0.1
https://github.com/gorakhargosh/watchdog/compare/v3.0.0...v4.0.0
https://github.com/gorakhargosh/watchdog/compare/v2.3.1...v3.0.0

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

mweinelt committed:
Bump watchdog to 6.0.0

MartinHjelmare commented on tests/components/zwave_js/fixtures/device_diagnostics.json:
```diff
@@ -20,7 +20,7 @@
     "pyscript": {
       "version": "1.5.0",
-      "requirements": ["croniter==1.3.8", "watchdog==2.3.1"]
+      "requirements": ["croniter==1.3.8", "watchdog==6.0.0"]
```
This is a test fixture that doesn't need updating. Please revert here.

mweinelt replied:
Good catch. Thank you!

edenhaus reviewed (dismissed):
Thanks for bumping the dependency! However, could you please update the PR description to contain at least one (or multiple) of the following:

- A link to the release notes of this package version, and all versions in between.
- A link to the changelog of this package.
- A link to a Git(Hub) diff/compare view from the current version to the bumped version.

This allows us to review upstream changes, which is needed to decide this change is working as intended and/or if we can include it in, for example, a patch release of Home Assistant.

Thanks already! 👍

mweinelt commented:
Updated the top post with the changelog and why the changes I made became necessary.

edenhaus commented:
Sanity check of `watchdog==6.0.0` :heavy_check_mark: 

Due the big version bump and several breaking changes, can you please test if the integration still works?

mweinelt commented:
![image](https://github.com/user-attachments/assets/d8595178-a0ff-40ce-8d1e-421792fda161)

Observed the following states:
- creation
- deletion
- moved
- modified (very shortly before going to closed)
- closed

frenck reviewed (approved):
Nice! Thanks, @mweinelt ❤️ 


../Frenck
